DigitalLearningSolutions.Data/Services/LearningHubSsoSecurityService.cs

```diff
@@ -56,6 +60,19 @@ public bool VerifyHash(string state, string hash)
             return false;
         }
 
+        private byte[] GetSalt(UTF8Encoding encoder)

```

Comment:
Nice idea to have a separate method. I don't think there's a benefit to parameterising the encoder, though. There's no reason to believe we'll use this method with a different kind of encoder (though it would be a good thing to parameterise if it were likely).
Could we change the name to make it more obvious that the salt is coming from the secret key? You don't even need to mention salt really - the method is just converting the string into bytes.